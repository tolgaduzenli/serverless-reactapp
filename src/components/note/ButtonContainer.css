.button-container{
    display:flex;
    flex-direction:row;
    justify-content: center;
    align-items: center;
}

.button-container > button{
    margin: 10px;
}