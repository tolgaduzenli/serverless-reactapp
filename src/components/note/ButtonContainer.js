import React from 'react'
import './ButtonContainer.css'
class ButtonContainer extends React.Component {
    render() {
        const { onClickDelete, onClickEdit } = this.props
        return <div className="button-container">
            <button className="waves-effect waves-light btn-small orange lighten-2" onClick={onClickEdit}>Edit</button>
            <button className="waves-effect waves-light btn-small red lighten-1" onClick={onClickDelete}>Delete</button>
        </div>
    }
}

export default ButtonContainer