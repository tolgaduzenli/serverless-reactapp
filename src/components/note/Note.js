import React from 'react'
import ButtonContainer from './ButtonContainer';

const Note = ({ note, onClickDelete, onClickEdit }) => <tr>
    <td>{note.title}</td>
    <td>{note.description}</td>
    <td>
        <ButtonContainer onClickDelete={onClickDelete} onClickEdit={onClickEdit} />
    </td>
</tr>

export default Note