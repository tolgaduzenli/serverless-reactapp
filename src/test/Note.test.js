import React from 'react'
import enzyme from 'enzyme'
import Note from '../components/note/Note'
import ButtonContainer from '../components/note/ButtonContainer'

describe('<Note />', () => {
    let wrapper
    let note = {
        title: "",
        description: "",
        _id: 1
    }
    
    beforeEach(() => {
        wrapper = enzyme.shallow(
            <Note key={""} note={note} onClickDelete={jest.fn()}
                onClickEdit={jest.fn()} />
        )
    })

    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('Should render a <tr />', () => {
        expect(wrapper.find('tr').length).toEqual(1)
    })

    it('should render a <ButtonContainer />', () => {
        expect(wrapper.find('td').containsMatchingElement(
            <ButtonContainer />
        )).toEqual(true)
    })
})