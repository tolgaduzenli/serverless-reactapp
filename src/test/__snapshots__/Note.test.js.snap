// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note
    note={
      Object {
        "_id": 1,
        "description": "",
        "title": "",
      }
    }
    onClickDelete={[MockFunction]}
    onClickEdit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          
        </td>,
        <td>
          
        </td>,
        <td>
          <ButtonContainer
            onClickDelete={[MockFunction]}
            onClickEdit={[MockFunction]}
          />
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ButtonContainer
            onClickDelete={[MockFunction]}
            onClickEdit={[MockFunction]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClickDelete": [MockFunction],
            "onClickEdit": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            
          </td>,
          <td>
            
          </td>,
          <td>
            <ButtonContainer
              onClickDelete={[MockFunction]}
              onClickEdit={[MockFunction]}
            />
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ButtonContainer
              onClickDelete={[MockFunction]}
              onClickEdit={[MockFunction]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClickDelete": [MockFunction],
              "onClickEdit": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
